src/com/aca/homework/week6/myinteger/MyInteger.java

```diff
@@ -0,0 +1,37 @@
+package com.aca.homework.week6.myinteger;
+
+public class MyInteger {
+    private static final MyInteger[] myIntegerObjects = new MyInteger[201];
+    private final int value;
+
+    private MyInteger(int value) {
+        this.value = value;
+    }
+
+    public static MyInteger valueOf(int value) {
+        checkNumberRange(value);
```

Comment:
when the value is out of the cache range, just create a new object without throwing an exception